src/components/molecules/FormEditor/FormEditor.tsx

```diff
@@ -122,25 +127,44 @@ const FormEditor = (props: {formSchema: any; formUiSchema?: any}) => {
 
   useDebounce(
     () => {
+      let formString = stringify(formData);
+
       if (selectedResource) {
-        let formString = stringify(formData);
         const content = mergeManifests(selectedResource.text, formString);
-
         if (content.trim() !== selectedResource.text.trim()) {
           dispatch(updateResource({resourceId: selectedResource.id, content}));
         }
+      } else if (selectedPath) {
+        try {
+          const filePath = getAbsoluteFilePath(selectedPath, fileMap);
```

Comment:
I think for future consideration we should have selectors for these types of functions so we don't have to import the entire `fileMap` state into each components that needs to call a simple function like this one.
By using custom selectors, only the output of the selector will affect the rendering of the component, while in this case, any fileMap change will trigger a re-render of this component